pkg/vault/test/foundry/Permit2.t.sol

```diff
@@ -42,67 +43,68 @@ contract Permit2Test is BaseVaultTest {
 
-        bytes[] memory permitSignatures = new bytes[](1);
-        (uint8 v, bytes32 r, bytes32 s) = getPermitSignature(
-            IEIP712(pool),
-            alice,
-            address(router),
-            bptAmountOut,
-            0,
-            block.timestamp,
-            aliceKey
-        );
-        permitSignatures[0] = abi.encodePacked(r, s, v);
+        (amount, , ) = permit2.allowance(alice, address(dai), address(router));
+        // Allowance is spent.
+        assertEq(amount, 0, "DAI allowance is not spent");
 
-        IAllowanceTransfer.PermitBatch memory permit2Batch = getPermit2Batch(
-            address(router),
-            [address(usdc), address(dai)].toMemoryArray(),
-            uint160(defaultAmount),
-            type(uint48).max,
-            0
-        );
+        (amount, , ) = permit2.allowance(alice, address(usdc), address(router));
+        // Allowance is spent.
+        assertEq(amount, 0, "USDC allowance is not spent");
+    }
+
+    function testPermitBatchAndCallDos() public {
+        (
+            IRouterCommon.PermitApproval[] memory permitBatch,
+            bytes[] memory permitSignatures,
+            IAllowanceTransfer.PermitBatch memory permit2Batch,
+            bytes memory permit2Signature,
+            bytes[] memory multicallData
+        ) = _setupMulticall();
 
-        bytes memory permit2Signature = getPermit2BatchSignature(
+        (uint160 amount, , ) = permit2.allowance(alice, address(usdc), address(router));
+        assertEq(amount, 0);
+        (amount, , ) = permit2.allowance(alice, address(dai), address(router));
+        assertEq(amount, 0);
+        assertEq(IERC20(pool).allowance(alice, address(router)), 0, "Router allowance is not zero");
+
+        // Bob needs to sign his own permit2.
+        permit2.allowance(bob, address(usdc), address(router));
+        permit2.allowance(bob, address(dai), address(router));
+
+        bytes memory permit2SignatureBob = getPermit2BatchSignature(
             address(router),
             [address(usdc), address(dai)].toMemoryArray(),
             uint160(defaultAmount),
             type(uint48).max,
             0,
-            aliceKey
+            bobKey
         );
 
-        bytes[] memory multicallData = new bytes[](2);
-        multicallData[0] = abi.encodeCall(
-            IRouter.addLiquidityUnbalanced,
-            (pool, amountsIn, bptAmountOut, false, bytes(""))
-        );
-
-        uint256[] memory minAmountsOut = [uint256(defaultAmount), uint256(defaultAmount)].toMemoryArray();
-        multicallData[1] = abi.encodeCall(
-            IRouter.removeLiquidityProportional,
-            (pool, bptAmountOut, minAmountsOut, false, bytes(""))
-        );
+        // Bob front-runs and calls using alice's `permitSignatures`.
+        vm.prank(bob);
+        router.permitBatchAndCall(permitBatch, permitSignatures, permit2Batch, permit2SignatureBob, multicallData);
```

Comment:
My understanding is that the only thing that Bob executes for Alice is the permit, granting Alice allowance. Can we add an assertion below checking that Bob gave Alice allowance using her signature? Also I don't think we need the `permit2Batch` and the multicall data, as that just makes Bob execute stuff for himself, and here we're just interested in the effect of Bob frontrunning Alice.

Reply:
This version has more direct tests (including one for the bubble up revert).